acme/acme.go

```diff
@@ -155,7 +196,8 @@ func (a *ACME) CreateClusterConfig(leadership *cluster.Leadership, tlsConfig *tl
 	}
 
 	a.store = datastore
-	a.challengeProvider = &challengeProvider{store: a.store}
+	a.challengeTLSProvider = &challengeTLSProvider{store: a.store}
+	a.challengeHTTPProvider = &challengeHTTPProvider{store: a.store}
```

Comment:
I'm not in love with instantiating `challengeTLSProvider` & `challengeHTTPProvider` here.